### PR TITLE
added yahoo stock data

### PR DESCRIPTION
**Pull request by OfficerHalf**

### CODE REVIEW

**Comment by ss2111** [comment]
*[YahooStockData/get_data.py, line 6 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIHdj38x0M_dRMkBWG:-KEIHdj38x0M_dRMkBWH:1955645752) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/get_data.py#L6)):*
There are no comments in any of the files and testing code appears to be left in as comments

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/get_data.py, line 8 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIHmhzxpXpAtkxyj33:-KEIHmhzxpXpAtkxyj34:-934340892) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/get_data.py#L8)):*
Only the Apple ticker? Should it not take input for the ticker?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/stock_data_window.py, line 12 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIJQxGa3XK9x_BLIHa:-KEIJQxGa3XK9x_BLIHb:-696935328) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/stock_data_window.py#L12)):*
Are the two commented lines needed? Is there a default? Remove the lines or use them.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/StockAnalysis/historical_bars.py, line 14 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIJo2f6S1li38JLaH0:-KEIJo2f6S1li38JLaH1:-812323123) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/StockAnalysis/historical_bars.py#L14)):*
Needs to make sure the path exists and have a less exact path

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/StockAnalysis/historical_bars.py, line 28 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIK1K-mMy2mRL_PKvm:-KEIK1K-mMy2mRL_PKvn:1098177869) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/StockAnalysis/historical_bars.py#L28)):*
What is 'ohlc'? Needs a better name or a comment to describe it.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/TradeStation/ts_check_controller.py, line 44 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIKJQFS40SpA5mXktl:-KEIKJQFS40SpA5mXktm:821453240) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/TradeStation/ts_check_controller.py#L44)):*
Test code.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/TradeStation/ts_requirements.py, line 5 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIKWXzU2Vk0lE6tvyt:-KEIKWXzU2Vk0lE6tvyu:1172473125) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/TradeStation/ts_requirements.py#L5)):*
Needs to check the file path again

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by ss2111** [comment]
*[YahooStockData/TradeStation/ts_requirements.py, line 2 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIKo3WXP45BAr4xkoh:-KEIKo3WXP45BAr4xkoi:1295020298) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/TradeStation/ts_requirements.py#L2)):*
Why import walk from os and os? Walk is only used once just use os.walk

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by OfficerHalf** [comment]
Reviewed 4 of 8 files at r1.
Review status: 4 of 8 files reviewed at latest revision, 12 unresolved discussions.

---

*[YahooStockData/key_statistics.py, line 4 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEILL4lb5iKH6fyMzeB:-KEILL4lb5iKH6fyMzeC:-1914810062) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/key_statistics.py#L4)):*
Should these headers be defined elsewhere, possibly a config file?

---

*[YahooStockData/key_statistics.py, line 42 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEILZ21eoyfF_fXGNVO:-KEILZ22yhq-d65oYZP2:-848942268) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/key_statistics.py#L42)):*
What is this section doing? Needs some documentation, especially for string wizardry.

---

*[YahooStockData/Apps-Trading/gap_analysis.py, line 6 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIM12LFsAr2cx6xHHW:-KEIM12LFsAr2cx6xHHX:-1859312222) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/gap_analysis.py#L6)):*
Should be defined in a config file.

---

*[YahooStockData/Apps-Trading/gap_analysis.py, line 29 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEILx0KWZlZ06xvYLCw:-KEILx0Ll9LE9x6Z4sOQ:1637081808) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/gap_analysis.py#L29)):*
Why is this percentage hardcoded?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by OfficerHalf** [comment]
Reviewed 3 of 8 files at r1.
Review status: 7 of 8 files reviewed at latest revision, 15 unresolved discussions.

---

*[YahooStockData/Apps-Trading/ts_check_view.py, line 3 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIMEfXRPUy4OkDDUGy:-KEIMEfYCtJyYTS79qdk:2031689466) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/ts_check_view.py#L3)):*
Why use PyQt earlier and tkinter now? Pick one and use it, there should be no need for both.

---

*[YahooStockData/StockAnalysis/historical_bars.py, line 10 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIMPto9VizrHvZGa9L:-KEIMPto9VizrHvZGa9M:-1447338639) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/StockAnalysis/historical_bars.py#L10)):*
Candlesticks? What does this variable do?

---

*[YahooStockData/TradeStation/ts_check_controller.py, line 15 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIM_blZZo2Efjy52vi:-KEIM_blZZo2Efjy52vj:121550136) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/TradeStation/ts_check_controller.py#L15)):*
Use of temp variables is not usually a good idea, especially since you use it so much.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by OfficerHalf** [comment]
Reviewed 1 of 8 files at r1.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by krislmassey** [comment]
*[YahooStockData/TradeStation/ts_requirements.py, line 5 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEISsi17yYBFDbrGlMi:-KEISsi17yYBFDbrGlMj:-1834595769) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/TradeStation/ts_requirements.py#L5)):*
More commented code in this file to get rid of

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*

**Comment by krislmassey** [comment]
Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

*[YahooStockData/get_data.py, line 17 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIIxYlvyKW_cmYjCNi:-KEIIxYlvyKW_cmYjCNj:51173351) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/get_data.py#L17)):*
All of this commented out code below looks like it should be removed.

---

*[YahooStockData/key_statistics.py, line 1 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEILMOP-dLfAZqi2iK7:-KEILMOP-dLfAZqi2iK8:-814716966) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/key_statistics.py#L1)):*
Renaming the file might help to better reflect what the YahooKeyStatistics class is accomplishing (parsing and extracting data).  

---

*[YahooStockData/Apps-Trading/gap_analysis.py, line 12 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIMS7YhkJC1i-IjUXw:-KEIMS7YhkJC1i-IjUXx:2082799040) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/gap_analysis.py#L12)):*
What is "candlesticks?"  This seems like bad variable naming.  Hard to follow what is being accomplished.  Either need a better name or a comment explaining this.

---

*[YahooStockData/Apps-Trading/gap_analysis.py, line 13 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIMNbxofl9VA1L-FrO:-KEIMNbxofl9VA1L-FrP:1895175693) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/gap_analysis.py#L13)):*
Ditch the commented out code.

---

*[YahooStockData/Apps-Trading/gap_analysis.py, line 31 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEINAdGwJLQEYAx9KBn:-KEINAdI6DMWnthZNEpM:1708568419) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/gap_analysis.py#L31)):*
More commented out code in need of removal.

---

*[YahooStockData/Apps-Trading/ts_check_view.py, line 22 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEINM0UubSwJLMqp714:-KEINM0UubSwJLMqp715:-499111021) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/Apps-Trading/ts_check_view.py#L22)):*
Use the code or get rid of the comment.

---

*[YahooStockData/StockAnalysis/historical_bars.py, line 22 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEINpICy2Xf9nE6LhWZ:-KEINpICy2Xf9nE6LhW_:484109535) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/StockAnalysis/historical_bars.py#L22)):*
A lot more commented out code here that needs cleaning up.

---

*[YahooStockData/StockAnalysis/historical_bars.py, line 42 \[r1\]](https://reviewable.io:443/reviews/officerhalf/codereview/1#-KEIO1yvmsl7D3onf363:-KEIO1yvmsl7D3onf364:-940717119) ([raw file](https://github.com/officerhalf/codereview/blob/fd33ac070fdc46b0f90c02ada2ca188678eed90f/YahooStockData/StockAnalysis/historical_bars.py#L42)):*
this is still a bad variable.  Needs a new name or explanation in all the files it is used in.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/officerhalf/codereview/1)*
